fedimint-server/src/config/mod.rs

```diff
@@ -133,8 +133,10 @@ pub struct ServerConfigConsensus {
     #[serde(with = "serde_tls_cert_map")]
     pub tls_certs: BTreeMap<PeerId, rustls::Certificate>,
     /// All configuration that needs to be the same for modules
+    pub modules: BTreeMap<ModuleInstanceId, ClientModuleConfig>,
```

Comment:
This is the core change. `modules` is now basically a `BTreeMap<ModuleInstanceId, (Kind, ConsensuVersion, Vec<u8>)`.

Everything else is to just make the compiler accept this new reality.

I leave `modules_json` like `modules` were before, to help the puny human operators that are not capable of decoding hex strings on the fly.